wire/msgblock.go

```diff
@@ -106,7 +106,19 @@ func (msg *MsgBlock) Deserialize(r io.Reader) error {
 	// At the current time, there is no difference between the wire encoding
 	// at protocol version 0 and the stable long-term storage format.  As
 	// a result, make use of BtcDecode.
-	return msg.BtcDecode(r, 0)
+	//
+	// Passing an encoding type of WitnessEncoding to BtcEncode for
+	// indicates that the transactions within the block are expected to be
+	// serialized according to the new serialization structure defined in
+	// BIP0141.
+	return msg.BtcDecode(r, 0, WitnessEncoding)
```

Comment:
Yeah when I was writing this originally, I had `Deserialize` always use the base encoding, and a special method for deserializing segwit blocks. In the end I reverted it as I concluded that eventually, non-segwit would be the exception. Doing it this way also results in less code changes across libraries that already utilize `wire.MsgBlock` and `wire.MsgTx` as they don't need to change all their interaction to call a new method. 

This _shouldn't_ cause reading existing blocks from disk to fail. That would only be the case if the block written to disk had a "zero txin" transaction, but IIRC those are invalid by consensus rules. 

The case that would cause reading exisging blocks from disk would be something like:
  * User upgrades to post-segwit version of `btcd`. 
  * User then downgrades to prior version of `btcd`, pre-segwit. 
  * `btcd` won't be able to start up if the latest or any prior blocks were written in the segwit encoding, as what it thinks is the `txin` count is something else entirely